AI #2117: Refactor format of CommitmentProgramEligibilityDetails (#2283)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/appendix/json_object_examples/commitment_program_eligibility_details_examples.md

> ## Examples: Commitment Program Eligibility Details

Reviewer: ```suggestion
# Examples: Commitment Program Eligibility Details
```

`allocated_method_details_examples.md`, `contract_applied_examples.md`, and `contract_commitment_applicability_examples.md` all start with `# Examples: [Name]` at H1. All four files are included via `!INCLUDE ...,1` in `json_object_examples.mdpp`, which bumps heading levels by one. Starting at H2 here causes this section's headings to render one level deeper than peer sections.

Author: Great point.  Addressed holistically in a separate commit.

The examples below are not exhaustive and may change over time. Service providers are the authoritative source for their [*commitment programs*](#glossary:commitment-program).

## Aura Web (Partially Covered Compute Usage)

Scenario: A compute usage row that is partially covered by a Flexible Spend Plan. The eligibility column still reflects all programs this usage qualifies for, regardless of current coverage.

| ServiceProviderName | ServiceName | CommitmentProgramEligibilityDetails                                                                            |
|---------------------|-------------|---------------------------------------------------------------------------------------------------------|

> | Aura Web            | Compute     | {"CommitmentPrograms": [{"ProgramType": "FlexibleSpendPlan"}, {"ProgramType": "ResourceReservation"}]} |

Reviewer: Aligning with Title Case direction for ProgramType examples (replayed from #2266).

```suggestion
| Aura Web            | Compute     | {"CommitmentPrograms": [{"ProgramType": "Flexible Spend Plan"}, {"ProgramType": "Resource Reservation"}]} |
```

## StackLens (Observability with Interval Spend Commitment)

Scenario: An observability platform usage row eligible for Monthly and Annual interval spend commitment pricing, offering lower effective rates than standard usage.

| ServiceProviderName | ServiceName   | CommitmentProgramEligibilityDetails                                                                                            |
|---------------------|---------------|-------------------------------------------------------------------------------------------------------------------------|

> | StackLens           | Observability | {"CommitmentPrograms": [{"ProgramType": "MonthlyIntervalSpendCommitment"}, {"ProgramType": "AnnualIntervalSpendCommitment"}]} |

Reviewer: Aligning with Title Case direction for ProgramType examples (replayed from #2266).

```suggestion
| StackLens           | Observability | {"CommitmentPrograms": [{"ProgramType": "Monthly Interval Spend Commitment"}, {"ProgramType": "Annual Interval Spend Commitment"}]} |
```

## LatticeScale (Ineligible Object Storage Usage)

Scenario: Standard object storage usage or a support fee, which is not eligible for any commitment program.

| ServiceProviderName | ServiceName   | CommitmentProgramEligibilityDetails |
|---------------------|---------------|------------------------------|
| LatticeScale        | ObjectStorage | null                         |

## Aura Web (Advance Resource Commitment-Eligible Compute Usage)

Scenario: A compute instance type and tenancy that are eligible for both discount-bearing programs and advance resource commitments. The eligibility column reflects all commitment constructs the usage qualifies for.

| ServiceProviderName | ServiceName | CommitmentProgramEligibilityDetails                                                                                                                                                                           |
|---------------------|-------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Aura Web            | Compute     | {"CommitmentPrograms": [{"ProgramType": "FlexibleSpendPlan"}, {"ProgramType": "ResourceReservation"}, {"ProgramType": "AdvanceResourceCommitment"}, {"ProgramType": "ZonalResourceCommitment"}]} |

Reviewer: Aligning with Title Case direction for ProgramType examples (replayed from #2266).

```suggestion
| Aura Web            | Compute     | {"CommitmentPrograms": [{"ProgramType": "Flexible Spend Plan"}, {"ProgramType": "Resource Reservation"}, {"ProgramType": "Advance Resource Commitment"}, {"ProgramType": "Zonal Resource Commitment"}]} |
```

## Coverage Rate with Eligibility-Adjusted Denominator

This example demonstrates how to calculate an accurate [*commitment*](#glossary:commitment) coverage rate using [CommitmentProgramEligibilityDetails](#datasets.costandusage.commitmentprogrameligibilitydetails) alongside [CommitmentDiscountId](#datasets.costandusage.commitmentdiscountid).

Acme Corp runs compute workloads on Aura Web. Some usage is covered by a Resource Reservation, some is eligible but uncovered, and a support fee is ineligible for any [*commitment program*](#glossary:commitment-program).

Three usage rows for a single charge period (2025-04-01):

1. **Uncovered compute** (Row 1): Eligible for FlexibleSpendPlan and ResourceReservation, not currently covered. [BilledCost](#datasets.costandusage.billedcost) and [EffectiveCost](#datasets.costandusage.effectivecost) are both &dollar;200.00.

> 2. **Covered compute** (Row 2): Covered by a ResourceReservation. CommitmentProgramEligibilityDetails is populated. BilledCost is &dollar;0.00; EffectiveCost is &dollar;150.00.

Reviewer: Aligning narrative to Title Case (replayed from #2266).

```suggestion
1. **Uncovered compute** (Row 1): Eligible for Flexible Spend Plan and Resource Reservation, not currently covered. [BilledCost](#datasets.costandusage.billedcost) and [EffectiveCost](#datasets.costandusage.effectivecost) are both &dollar;200.00.
2. **Covered compute** (Row 2): Covered by a Resource Reservation. CommitmentProgramEligibilityDetails is populated. BilledCost is &dollar;0.00; EffectiveCost is &dollar;150.00.
```

Acme Corp runs compute workloads on Aura Web and uses StackLens for observability monitoring. Some Aura Web compute usage is covered by a Resource Reservation. StackLens usage is uncovered but eligible for Interval Spend Commitments at monthly or annual terms. A practitioner wants to answer: "Which *commitment program* and provider should I target for my next purchase?"

Six usage rows for a single charge period (2025-04-01):

1. **Uncovered compute** (Rows 1-2): Two Aura Web Compute rows eligible for both FlexibleSpendPlan and ResourceReservation. [BilledCost](#datasets.costandusage.billedcost) totals &dollar;500.00 across both rows.
2. **Covered compute** (Row 3): Aura Web Compute covered by an existing ResourceReservation. Filtered out by the query because [CommitmentDiscountId](#datasets.costandusage.commitmentdiscountid) is populated.
3. **Ineligible support** (Row 4): Aura Web Support with no CommitmentProgramEligibilityDetails. Filtered out because the column is null.

> 4. **Uncovered observability** (Rows 5-6): Two StackLens Observability rows eligible for MonthlyIntervalSpendCommitment and AnnualIntervalSpendCommitment. BilledCost totals &dollar;200.00.

Reviewer: Aligning narrative to Title Case (replayed from #2266).

```suggestion
1. **Uncovered compute** (Rows 1-2): Two Aura Web Compute rows eligible for both Flexible Spend Plan and Resource Reservation. [BilledCost](#datasets.costandusage.billedcost) totals &dollar;500.00 across both rows.
2. **Covered compute** (Row 3): Aura Web Compute covered by an existing Resource Reservation. Filtered out by the query because [CommitmentDiscountId](#datasets.costandusage.commitmentdiscountid) is populated.
3. **Ineligible support** (Row 4): Aura Web Support with no CommitmentProgramEligibilityDetails. Filtered out because the column is null.
4. **Uncovered observability** (Rows 5-6): Two StackLens Observability rows eligible for Monthly Interval Spend Commitment and Annual Interval Spend Commitment. BilledCost totals &dollar;200.00.
```

To evaluate these purchasing options, the `CommitmentPrograms` JSON array must be flattened so each eligible program can be analyzed independently. By expanding the array (e.g., via `CROSS JOIN UNNEST`) and grouping the uncovered costs by `ServiceProviderName`, `ServiceName`, and `EligibleProgramType`, the practitioner gets the summary presented in the table below. A reference implementation is available in the [eligible uncovered spend query](#supportedfeatures.commitmentprogrameligibilitydetails) supported feature.

| ServiceProviderName | ServiceName | EligibleProgramType | EligibleUncoveredCost |
|:--------------------|:------------|:--------------------|:---------------------------|
| Aura Web | Compute | FlexibleSpendPlan | &dollar;500.00 |
| Aura Web | Compute | ResourceReservation | &dollar;500.00 |
| StackLens | Observability | MonthlyIntervalSpendCommitment | &dollar;200.00 |

> | StackLens | Observability | AnnualIntervalSpendCommitment | &dollar;200.00 |

Reviewer: Aligning output table EligibleProgramType values to Title Case (replayed from #2266).

```suggestion
| Aura Web | Compute | Flexible Spend Plan | &dollar;500.00 |
| Aura Web | Compute | Resource Reservation | &dollar;500.00 |
| StackLens | Observability | Monthly Interval Spend Commitment | &dollar;200.00 |
| StackLens | Observability | Annual Interval Spend Commitment | &dollar;200.00 |
```

> Aura Web Compute appears as &dollar;500.00 under both FlexibleSpendPlan and ResourceReservation. This does not mean &dollar;1,000.00 is uncovered. The &dollar;500.00 is the same spend, and each program type represents an independent purchasing opportunity. Purchasing a FlexibleSpendPlan would cover some or all of that &dollar;500.00, as would a ResourceReservation. The practitioner must choose between them (or split across both) based on flexibility requirements and discount depth.

Reviewer: Aligning narrative to Title Case (replayed from #2266).

```suggestion
Aura Web Compute appears as &dollar;500.00 under both Flexible Spend Plan and Resource Reservation. This does not mean &dollar;1,000.00 is uncovered. The &dollar;500.00 is the same spend, and each program type represents an independent purchasing opportunity. Purchasing a Flexible Spend Plan would cover some or all of that &dollar;500.00, as would a Resource Reservation. The practitioner must choose between them (or split across both) based on flexibility requirements and discount depth.
```
